src/formik-validation.spec.ts

```diff
@@ -1453,6 +1453,55 @@ when adding one validator to a given nested field with kebap case
         expect(myFieldValidation1).toHaveBeenCalled();
         expect(myFieldValidation2).toHaveBeenCalled();
         expect(validationResult).toBeNull();
+
+        done();
+      });
+    });
+
+    it(`#Spec 14.1: should failed form validation, and return one field validation result
+    and form validation result
+    when adding two fields validation and one of them fails with nested fields
+    (return expected formik structure for error reporting)
+    `, done => {
+      // Arrange
+      const myFieldValidation1: FieldValidationFunctionSync = jest.fn(
+        fieldValidatorArgs => ({
+          type: 'MY_TYPE_A',
+          succeeded: true,
+          message: 'mymessageA',
+        })
+      );
+      const myFieldValidation2: FieldValidationFunctionSync = jest.fn(
+        fieldValidatorArgs => ({
+          type: 'MY_TYPE_B',
+          succeeded: false,
+          message: 'mymessageB',
+        })
+      );
+
+      const validationSchema: ValidationSchema = {
+        field: {
+          'nested.field1': [myFieldValidation1],
+          'nested.field2': [myFieldValidation2],
+        },
+      };
+
+      const values = {};
+
+      // Act
+      const formValidation = createFormikValidation(validationSchema);
+      const result = formValidation.validateForm(values);
+
+      // Assert
+      result.then(validationResult => {
+        expect(myFieldValidation1).toHaveBeenCalled();
+        expect(myFieldValidation2).toHaveBeenCalled();
+        expect(validationResult).toEqual({
+          nested: {
+            field1: '',
```

Comment:
I'm not sure why this field value is an empty string, AFAIK formik will consider this as an error too

Reply:
Just making a quick test on this sandbox: https://codesandbox.io/s/github/lemoncode/fonk/tree/master/examples/formik/js/basic 

Empty string seems to be ignored by Formik, but if it's better to stick with Formik we can remove the field.

My idea of informing this field as empty string was avoiding the consumer of this library having to check if error.field was informed, but I think it can be cleaner the way you propose, I would update:

```diff
private mapErrorsToFormikErrorMessageStructure = (errors: {
    [fieldId: string]: ValidationResult;
  }): Record<string, string> => {
    const formikErrors = {};

    for (const key of Object.keys(errors)) {
      const errorMessage =
-        errors[key] && !errors[key].succeeded ? errors[key].message : '';
+        errors[key] && !errors[key].succeeded ? errors[key].message : null;
+ if (errorMessage) {
      set(formikErrors, key, errorMessage);
+  }
    }

    return formikErrors;
  };
```

Reply:
please check this one out: https://codesandbox.io/s/validate-field-bhy2g
even with `null` as value formik does not submit the form. 

if there's not error, key should not set in errors object

Reply:
Thanks @sijad, updated the pull, now field errors keys are not set in error object if no error, and record entry won't exists as well if there aren't record validations or all record validations succeeded